src/lib.rs

```diff
@@ -14,6 +14,82 @@ lazy_static::lazy_static! {
     static ref GLOBAL: Arc<Global> = Arc::new(Global::new("wgpu", wgc::hub::IdentityManagerFactory));
 }
 
+/// Follow a chain of next pointers and automatically resolve them to the underlying structs.
+///
+/// # Syntax:
+///
+/// Given:
+///
+/// `fn map_thing_descriptor(base: &ThingDescriptor, ext1: Option<&ThingDescriptorExtension1>) -> wgt::ThingDescriptor`
+///
+/// Use the syntax:
+///
+/// `follow_chain!(map_thing_descriptor(base_c_descriptor, ThingDescriptorExtension1STypeValue => ThingDescriptorExtension1))`
+///
+/// # Safety
+///
+/// This macro does not use any internal unsafe blocks. The caller (or most likely the function) needs
+/// to be unsafe. The following constraints must be upheld for this to be valid:
+///
+/// - All pointers in the chain of next pointers must point to either null or a valid extension object
+/// - All non-null pointers must be sufficiently aligned such as they can be dereferenced as both a [`ChainedStruct`]
```

Comment:
this line will then be possible to remove, since we panic otherwise (or can stop traversing if panic is not desired)